src/lib/addProvide.js

```diff
@@ -0,0 +1,21 @@
+function addProvide (component, options) {
+  const provide = typeof options.provide === 'function'
```

Comment:
Had to test here for if the provide is a function, otherwise the provide was turning into what appears to be a spy.

![2017-06-05 at 9 30 pm](https://cloud.githubusercontent.com/assets/1709860/26811392/2d366d06-4a37-11e7-8d4e-0b4459f93805.png)